fix: showLineNumbers{n} with Multiple theme bug

Likilee opened the pull request:
## Related Issue
There is an issue with the functionality of specifying the starting line number when using `showLineNumbers{n}` with Multiple Themes. [#80](link-to-issue)

## Root Cause Analysis
In the case of applying multiple themes, the `index` of the code element node with the second theme is 1. Therefore, it fails to pass the condition `if (index === 0)`, resulting in the failure of the logic that sets the counter-set as an inline style.

## Problem Resolution
The condition `if (index === 0)` has been removed. It has been verified that the tests are passing.
![image](https://github.com/atomiks/rehype-pretty-code/assets/35288028/288f8680-b00a-4641-a364-c4a3189d38db)

## Request for Feedback
Are there any potential issues or concerns with removing the condition `if (index === 0)`?

atomiks commented:
Thank you 🙏 can you add a test that fails without the change (i.e. add another theme block)

Likilee commented:
Hi @atomiks 😄  As requested, I have added the test case. 
During the process, for test cases that use multiple themes, I added logic to the test code that allows changing the theme in the settings.

I have a question. It seems that the getHighlighter function is called three times when running with multiple themes. Is it correct for the `expect(getHighlighter).toHaveBeenCalledTimes(1);` part to work the same way even for multiple themes?

Here is the result of the test execution.
![image](https://github.com/atomiks/rehype-pretty-code/assets/35288028/44735b97-2ea4-457f-83e6-51612bb8aeb5)

atomiks commented:
The highlighter needs to be called for each different theme so it looks expected and needs to change to adjust for that

Likilee commented:
I have made modifications to the test script based on your feedback. I have separated the test script into Single Theme test suite and Multiple Theme test suite. Would this be acceptable?
![image](https://github.com/atomiks/rehype-pretty-code/assets/35288028/d34dc65b-e118-4b8b-bf96-99d5903167ce)

atomiks commented:
Thank you!
